### ISSUE TITLE
Settings snippet editor: improve fields focus style and blur

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#619]
In https://github.com/Yoast/yoast-components/pull/542 we added an `onBlur` prop and an `unsetFieldFocus` callback to remove the highlight that emulates the focus style from the editor. This is required as, without it, the "active" field still looks like focused when it loses focus.

We missed to do the same thing for the `SettingsSnippetEditor` which misses equivalent prop and function. Right now, when tabbing away or clicking outside one of the editors in the Settings pages, it still looks "focused":

<img width="717" alt="screen shot 2018-06-25 at 15 00 43" src="https://user-images.githubusercontent.com/1682452/41851993-105e8560-788a-11e8-8b55-b02d15c23f2f.png">

Also, in wordpress-seo there's at least one plugin-specific implementation that doesn't have any focus style and needs some help. See the "Special pages" editors in the Search appearance > Archives tab.

<img width="702" alt="screen shot 2018-06-25 at 15 12 52" src="https://user-images.githubusercontent.com/1682452/41852099-6f99db56-788a-11e8-91d0-dc3979834115.png">

**Comment by maartenleenders** [comment]
ah woops. I may have accidentally boyscouted this in https://github.com/Yoast/yoast-components/pull/621/files

Should I split it out from there?

**Comment by boblinthorst** [comment]
CR: This one is fine, but placed a comment in the sibling pr, and this one shouldn't be acceptance tested without that one, so moved it back for process reasons.
